junit-platform-commons/src/main/java/org/junit/platform/commons/util/ClassNameFilterUtil.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.platform.commons.util;
+
+import static org.apiguardian.api.API.Status.INTERNAL;
+
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import org.apiguardian.api.API;
+
+/**
+ * Class-related predicate holder used by execution listener, execution condition predicates
+ *
+ * <h3>DISCLAIMER</h3>
+ *
+ * <p>These utilities are intended solely for usage within the JUnit framework
+ * itself. <strong>Any usage by external parties is not supported.</strong>
+ * Use at your own risk!
+ *
+ */
+@API(status = INTERNAL, since = "5.5")
+public class ClassNameFilterUtil {
+
+	private ClassNameFilterUtil() {
+		/* no-op */
+	}
+	private static final Predicate<?> alwaysActivated = object -> true;
+	private static final Predicate<?> alwaysDeactivated = object -> false;
+
+	public static final String DEACTIVATE_ALL_PATTERN = "*";
+	public static final String DEACTIVATE_CONDITIONS_PATTERN_PROPERTY_NAME = "junit.jupiter.conditions.deactivate";
+	public static final String DEACTIVATE_LISTENERS_PATTERN_PROPERTY_NAME = "junit.platform.execution.listeners.deactivate";
+
+	public static Predicate<?> get(Supplier<Optional<String>> pattern) {
```

Comment:
❓ Why a `Supplier`?

Reply:
Wanted to circumnavigate using an Optional as a method parameter along with trying to remove the dependency on Configuration parameter.

Using Optional as method parameters (or class members) isn't really ideal since they take only delegate a state of the variable while exposing a deceiving NPE potential on themselves. 

Do you suggest any other way?

Reply:
Can we just take a `String` and handle the `Optional` on the outside? If that ends up being too much duplication, I think `Optional<String>` would be better here and we don't have a general rule in this project (even though IntelliJ might disagree) against `Optional` as method parameter. A `Supplier` can be `null`, too. 😉